### ISSUE TITLE
metaschema/schema/object validation issue

### ISSUE THREAD

**Issue by jsher001** [everit-org/json-schema#217]
Hi

I'm trying to figure out how to do the following...
1. meta schema based on draft07 (extends using same approach as hyper schema)
2. schema based on this meta schema - can validate against the meta schema
3. object based on the schema - can validate against the schema

meta schema
```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "http://myschema.org/mymetaschema#",
  "title": "My Schema",
  "definitions": {
    "schemaArray": {
      "allOf": [ {
          "$ref": "http://json-schema.org/draft-07/schema#/definitions/schemaArray"
        }, {
          "items": { "$ref": "#" }
        } ]
    }
  },
  "allOf": [ { "$ref": "http://json-schema.org/draft-07/schema#" } ],
  "properties": {
    "additionalItems": { "$ref": "#" },
    "additionalProperties": { "$ref": "#" },
    "dependencies": {
      "additionalProperties": {
        "anyOf": [ { "$ref": "#" }, { "type": "array" } ]
      }
    },
    "items": {
      "anyOf": [ { "$ref": "#" }, { "$ref": "#/definitions/schemaArray" } ]
    },
    "definitions": { "$ref": "#" },
    "patternProperties": { "$ref": "#" },
    "properties": {
      "additionalProperties": { "$ref": "#" }
    },
    "if": { "$ref": "#" },
    "then": { "$ref": "#" },
    "else": { "$ref": "#" },
    "allOf": { "$ref": "#/definitions/schemaArray" },
    "anyOf": { "$ref": "#/definitions/schemaArray" },
    "oneOf": { "$ref": "#/definitions/schemaArray" },
    "not": { "$ref": "#" },
    "contains": { "$ref": "#" },
    "propertyNames": { "$ref": "#" },
    "ext1": { "type": "string" },
    "ext2": {
      "type": "object",
      "properties": {
        "foo": { "type": "string" }
      }
    }
  }
}
```

schema
```
{
  "$schema": "http://myschema.org/mymetaschema#",
  "$id": "http://myschema.org/myschema#",
  "title": "My schema",
  "description": "My schema",
  "properties": {
    "prop1": {
      "type": "string",
      "ext1": "xxx",
      "ext2": {
        "foo": "bar"
      }
    }
  }
}
```

object
```
{
	"prop1" : "foo"
}
```

Simple code to try both of the validations listed in 2. and 3. above.
```
package org.myschema;

import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;
import java.nio.file.Files;
import java.nio.file.Paths;

import org.everit.json.schema.Schema;
import org.everit.json.schema.loader.SchemaLoader;
import org.json.JSONObject;
import org.json.JSONTokener;

public class JsonMetaSchemaTester {
	
	private void validate(String schemaFilePath, String objectFilePath) throws Exception {
		System.out.println("-------------------------------------------");
		System.out.println(schemaFilePath);
		File schemaFile = new File(schemaFilePath);
		InputStream schemaInputStream = new FileInputStream(schemaFile);
		JSONObject rawSchema = new JSONObject(new JSONTokener(schemaInputStream));
		
		Schema schema = SchemaLoader.load(rawSchema);

		String object = new String(Files.readAllBytes(Paths.get(objectFilePath)));
		schema.validate(new JSONObject(object));
		System.out.println("Validation OK");
	}
	
    public static void main( String[] args ) {
    	try {
        	JsonMetaSchemaTester tester = new JsonMetaSchemaTester();
        	
        	// Validate schema against meta-schema
        	tester.validate("src/main/resources/my-meta-schema.json", "src/main/resources/my-schema.json");
        	
        	// Validate object against schema
        	tester.validate("src/main/resources/my-schema.json", "src/main/resources/my-object");
    	}
		catch (Exception e) {
			System.err.println(e.getMessage());
	    }
    }
}
```

First validation succeeds. 
Second fails...
```
-------------------------------------------
src/main/resources/my-meta-schema.json
Validation OK
-------------------------------------------
src/main/resources/my-schema.json
#: could not determine schema version: no meta-schema is known with URL [http://myschema.org/mymetaschema#]
```

Interestingly this works on https://www.jsonschemavalidator.net/

![Image1](http://i66.tinypic.com/6ht505.png)

![Image2](http://i63.tinypic.com/5bbm7a.png)

Hints or help appreciated. 

Thanks
   James

**Comment by jsher001** [comment]
Great stuff @erosb . Works a treat. Thanks for turning around fix so quickly.

A follow up question... would it be possible to release 1.9.2 with this ? I'd like to roll this out to my colleagues.

Regards
   James

**Comment by jsher001** [comment]
Gentle reminder @erosb . Any plans to release 1.9.2 ? It would be very helpful for me to promote use of this excellent project with my colleagues.
Regards
    James

**Comment by erosb** [comment]
@jsher001 what is the reason of needing 1.9.2 ? You can simply put `239816f1e214dbe66253c0db6fe8f81c6c07e871` into the <version> tag in your pom.xml , and JitPack will understand it, so you can get the change. So technically - assuming that you use maven or gradle - nothing stops you from sharing the fixed version with your colleagues.

We can release 1.9.2 if it is absolutely necessary for you, but I would more prefer implementing custom keyword support before the next release (which would be 1.10.0). That would happen at some time in September.

**Comment by jsher001** [comment]
Thanks for reply @erosb and for clarifying your release plans.
Since it's due this month I can wait for 1.10.0
Looking forward to trying it out.

Regards
   James

**Comment by erosb** [comment]
Hello @jsher001 I published version 1.9.2 with the fix (couldn't make much progress with the custom keyword support and also some other things are in progress, so better to release just a smaller maintenance release for now).

**Comment by jsher001** [comment]
Many thanks for the update @erosb
